ext/dom/html_document.c

```diff
@@ -1358,4 +1358,29 @@ zend_result dom_html_document_encoding_write(dom_object *obj, zval *newval)
 	return SUCCESS;
 }
 
+/* https://html.spec.whatwg.org/#dom-document-body */
+zend_result dom_html_document_body_read(dom_object *obj, zval *retval)
+{
+	DOM_PROP_NODE(const xmlDoc *, docp, obj);
+
+	const xmlNode *root = xmlDocGetRootElement(docp);
+	if (root == NULL || !(php_dom_ns_is_fast(root, php_dom_ns_is_html_magic_token) && xmlStrEqual(root->name, BAD_CAST "html"))) {
+		ZVAL_NULL(retval);
+		return SUCCESS;
+	}
+
+	xmlNodePtr cur = root->children;
+	while (cur != NULL) {
+		if (cur->type == XML_ELEMENT_NODE && php_dom_ns_is_fast(cur, php_dom_ns_is_html_magic_token)
+			&& (xmlStrEqual(cur->name, BAD_CAST "body") || xmlStrEqual(cur->name, BAD_CAST "frameset"))) {
```

Comment:
I'm confused by the `"frameset"` here.

Reply:
From https://html.spec.whatwg.org/#dom-document-body

> The body attribute, on getting, must return [the body element](https://html.spec.whatwg.org/#the-body-element-2) of the document (either a [body](https://html.spec.whatwg.org/#the-body-element) element, a [frameset](https://html.spec.whatwg.org/#frameset) element, or null).

i.e. in a document with frames the frameset element is the body instead of body.

Reply:
TIL, also didn't know about it as it has been deprecated for seemingly ages.